src/components/plots/PlotContainer.jsx

```diff
@@ -0,0 +1,147 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch, useSelector } from 'react-redux';
+
+import {
+  Collapse, Button, Skeleton, Space, Tooltip,
+} from 'antd';
+import { InfoCircleOutlined } from '@ant-design/icons';
+
+import { LOAD_CONFIG } from 'redux/actionTypes/componentConfig';
+
+import { savePlotConfig } from 'redux/actions/componentConfig/index';
+import { initialPlotConfigStates } from 'redux/reducers/componentConfig/initialState';
+import _ from 'lodash';
+
+const { Panel } = Collapse;
+
+const ResetButton = ({ onClickReset, disabled }) => (
+  <Button
+    key='reset'
+    type='primary'
+    size='small'
+    onClick={onClickReset}
+    disabled={disabled}
+  >
+    Reset
+  </Button>
+);
+
+ResetButton.propTypes = {
+  onClickReset: PropTypes.func.isRequired,
+  disabled: PropTypes.bool.isRequired,
+};
+
+const PlotContainer = (props) => {
+  const {
+    experimentId,
+    plotUuid, plotType, plotInfo,
+    extra, showReset, children,
+  } = props;
+
+  const dispatch = useDispatch();
+
+  const [enableReset, setEnableReset] = useState(true);
+  const { config } = useSelector((state) => state.componentConfig[plotUuid] || {});
+
+  const checkConfigEquality = (currentConfig, initialConfig) => {
+    const ignoredFields = {
+      // config fields that are set dynamically on component render
+      // should not be compared to their initial values
+      embeddingContinuous: ['shownGene'],
+      violin: ['shownGene', 'title'],
+      markerHeatmap: ['selectedGenes'],
+      DotPlot: ['selectedGenes'],
+    };
+
+    const areAllValuesTheSame = Object.keys(initialConfig).every((key) => {
+      // By pass plot data because we want to compare settings not data
+      if (key === 'plotData') return true;
+      if (ignoredFields[plotType]?.includes(key)) return true;
+      if (typeof currentConfig[key] === 'object') {
```

Comment:
I think `instanceof` can be used in this case, as all objects are derived from the `Object` object. However, semantically, I think `typeof` is better in this case, because we don't really care which object the value is derived from.